hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/OneReplicaPipelineSafeModeRule.java

```diff
@@ -75,69 +66,59 @@ public OneReplicaPipelineSafeModeRule(String ruleName, EventQueue eventQueue,
             HDDS_SCM_SAFEMODE_ONE_NODE_REPORTED_PIPELINE_PCT  +
             " value should be >= 0.0 and <= 1.0");
 
+    // Exclude CLOSED pipeline
     int totalPipelineCount =
         pipelineManager.getPipelines(HddsProtos.ReplicationType.RATIS,
-            HddsProtos.ReplicationFactor.THREE).size();
+            HddsProtos.ReplicationFactor.THREE, Pipeline.PipelineState.OPEN)
+            .size() +
+            pipelineManager.getPipelines(HddsProtos.ReplicationType.RATIS,
```

Comment:
I guess the original idea of this OneReplicaPipelineSafeModeRule is to check THREE factor Pipeline only since ONE factor pipeline is not favored in production environment.

Reply:
OneReplicaPipelineSafeModeRule as per the description tracks the open containers in the system so that atleast one replica is available. So I guess if we are tracking one node pipelines, we can track them in this rule as well?